core/src/main/java/org/infinispan/interceptors/ClusteredCacheLoaderInterceptor.java

```diff
@@ -41,24 +41,35 @@ private void startClusteredCacheLoaderInterceptor() {
    }
 
    @Override
-   protected boolean skipLoadForWriteCommand(WriteCommand cmd, Object key, InvocationContext ctx) {
+   protected boolean skipLoadForWriteCommand(WriteCommand cmd, Object key, InvocationContext ctx, boolean functionalReadWrite) {
```

Comment:
I'm not sure I like this boolean parameter either. It makes the method quite complex adding further if statements and it's already quite difficult to read. I think i'd rather add some code duplication by having functional-specific methods which are more focused, smaller, and easier to read.

Same thing for other methods that have been extended to have `functionalReadWrite` boolean parameters